Components/ProductToShop.php

```diff
@@ -433,6 +435,20 @@ public function insertOrUpdate(Product $product)
 
         $this->manager->flush();
 
+        //article has to be flushed
+        $categories = $this->categoryResolver->resolve($product->categories);
+        $hasMappedCategory = count($categories) > 0;
+        $detailAttribute->setConnectMappedCategory($hasMappedCategory);
```

Comment:
there isn't persist of ```detailAttribute```. Also flush was before this change.
If you see on some line below we call ```$this->manager->clear();```.
So this change won't be stored in db.